[chew01] iP

chew01 opened the pull request:
### Ducky
> "A very interesting (not) clone."  – the developer himself

Ducky helps you remember things to do. It's...
- text-based
- easy to learn
- ~~FAST~~ *SUPER* FAST to use!

All you need to do is...

1. download it from [here](https://github.com/chew01/ip/releases)
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you :wink:

And it is **FREE**!

Features:

- [x] Managing tasks
- [x] Managing deadlines
- [ ] Reminders (?)

If you're a Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

chew01 committed:
skeletal version of ducky

chew01 committed:
add std input loop

chew01 committed:
separate ui class

chew01 committed:
add storage and switch case to loop

chew01 committed:
add task class and changed strings to tasks

chew01 committed:
added command types and parser

chew01 committed:
add mark and unmark feature

chew01 committed:
created subtask classes

chew01 committed:
added subtask to parser and ui

chew01 committed:
added text ui testing

chew01 committed:
added invalid command, invalid task format exceptions

chew01 committed:
add trim to parser

chew01 committed:
add invalid index and number format exception

chew01 committed:
cleanup on exceptions

chew01 committed:
update storage to use arraylist

chew01 committed:
added delete feature

chew01 committed:
Add save data format string to Task classes

chew01 committed:
Add tasks.txt to .gitignore

chew01 committed:
Add save and load to disk methods to Storage class

chew01 committed:
Add save and load functionality to UserInterface and Ducky classes

chew01 committed:
Merge branch 'branch-Level-7'

chew01 committed:
Change DeadlineTask class to use LocalDate instead of String for deadline

chew01 committed:
Adapt Storage and UserInterface to use LocalDate for deadline

chew01 committed:
Change DeadlineTask format in list

chew01 committed:
Merge branch 'branch-Level-8'

chew01 committed:
Decouple TaskList from Storage class

chew01 committed:
Fix Storage class to be compatible with TaskList class

chew01 committed:
Add DuckyException as superclass for all existing exception classes

chew01 committed:
Refactor Ducky.java to separate command concerns from UserInterface class

chew01 committed:
Add helper functions to TaskList and UserInterface

chew01 committed:
Change exception text and fix bug in TaskList class

chew01 committed:
Add Command subclasses and changed Parser to use subclasses

chew01 committed:
Merge branch 'branch-A-MoreOOP'

chew01 committed:
Sort classes into packages

chew01 committed:
Merge branch 'branch-A-Packages'

chew01 committed:
Merge branch 'add-gradle-support'

chew01 committed:
Edit build.gradle file

chew01 committed:
Change Task class to abstract class

chew01 committed:
Add tests for all Task classes

chew01 committed:
Merge branch 'branch-A-JUnit'

chew01 committed:
Change JAR archive name

chew01 committed:
Move file line parsing to Parser class

chew01 committed:
Add javadoc to classes in main package

chew01 committed:
Add javadoc to task classes and parser

chew01 committed:
Add javadoc to command classes

chew01 committed:
Tweak code to comply to coding standard

chew01 committed:
Add FindTaskCommand class

chew01 committed:
Fix string formatting

chew01 committed:
Add find task command to Parser

chew01 committed:
Change implementation of find to add indices

chew01 committed:
Merge branch 'branch-A-JavaDoc'

chew01 committed:
Merge branch 'branch-A-CodingStandard'

chew01 committed:
Merge branch 'branch-Level-9'

CelestineTan03 commented on src/main/java/ducky/task/Task.java:
```diff
@@ -31,6 +31,15 @@ public void setIncomplete() {
+    /**
+     * Returns whether task description contains the given string as a substring.
+     * @param s String to check in description.
+     * @return True if description contains specified string.
```
To be more accurate, can change True to true

CelestineTan03 commented on src/main/java/ducky/Ducky.java:
```diff
@@ -0,0 +1,54 @@
+    private final UserInterface ui;
+
+    /**
+     * Constructs a Ducky chatbot instance with the specified file path for persistent data.
```
Good javadoc, remember to add an empty line between description and parameter section

CelestineTan03 commented on src/main/java/ducky/util/Parser.java:
```diff
@@ -0,0 +1,168 @@
+            // Check if there are 2 arguments
+            if (deadlineParts.length < 2) {
+                throw new DuckyInvalidCommandFormatException(
+                        "A description and deadline (in yyyy-mm-dd format) " +
```
Good indentation for wrapped lines. But better to break before an operator (eg "+").

CelestineTan03 commented on src/main/java/ducky/command/Command.java:
```diff
@@ -0,0 +1,29 @@
+     * Returns whether command is an exit command.
+     * @return True if command is an exit command.
+     */
+    public boolean isExit() {
```
Good method name for a boolean

CelestineTan03 commented on src/main/java/ducky/util/Parser.java:
```diff
@@ -0,0 +1,168 @@
+        String commandType = parts[0].toLowerCase();
+        String argumentString = (parts.length > 1) ? parts[1].trim() : "";
+
+        switch (commandType) {
```
Good style for switch statement

CelestineTan03 reviewed (commented):
Overall good coding standard, just a few minor improvements

chew01 committed:
Add Main and MainWindow GUI class

chew01 committed:
Merge branch 'branch-Level-10'

chew01 committed:
Add checkstyle configuration

chew01 committed:
Merge branch 'branch-A-CheckStyle'

chew01 committed:
Migrate from CLI to GUI

chew01 committed:
Remove old UI class

chew01 committed:
Update add task class

chew01 committed:
Update other task classes

chew01 committed:
Change package layout

chew01 committed:
Clean up files

chew01 committed:
Remove exit command

chew01 committed:
Rename Duke references

chew01 committed:
Add images

chew01 committed:
Merge branch 'branch-Level-10'

chew01 committed:
Add assertions to Ducky and TaskList

There is a possibility that the file path given to Ducky or a task added
to the TaskList is empty or null respectively.

Making sure invariants for both cases are held improves safety especially
if future commits allow file path or format to be changed.

Let's add assertions for both cases.

chew01 committed:
Merge pull request #1 from chew01/branch-A-Assertions

Add assertions to Ducky and TaskList

chew01 committed:
Rename folders

chew01 committed:
Extract constants from functions

The majority of chatbot responses are inline string responses.

Using string constants at the top of their classes is more
developer-friendly as they can be given readable names and also easier
to change as required.

Let's extract the current inline strings as constants.

chew01 committed:
Extract body of Parser switch case into handlers

The body of certain cases within Parser.parse() switch case is much
longer than other cases.

Since these cases tend towards arrowhead pattern, it would be better to
abstract these out as separate handler methods for readability and code
quality.

Let's extract these bodies out as methods and shift them into a separate
handler class.

chew01 committed:
Merge pull request #2 from chew01/branch-A-CodeQuality

Branch a code quality

chew01 committed:
Update build.gradle for javafx

chew01 committed:
Add view schedule command

chew01 committed:
Add schedule command to parser

chew01 committed:
Merge pull request #3 from chew01/branch-B-ViewSchedules

Branch b view schedules

chew01 committed:
Change task formatting

chew01 committed:
Add task list index spacing

chew01 committed:
Add Ui.png

chew01 committed:
Update README

chew01 committed:
Merge branch 'branch-A-UserGuide'

chew01 committed:
Update boolean naming to convention

chew01 committed:
Change github pages theme

chew01 committed:
Add line break for javadoc comments

chew01 committed:
Edit readme
